src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -87,49 +100,55 @@ private static void MigrateRuntimes(PackageSpec packageSpec, Microsoft.Build.Eva
             if (runtimes != null && runtimes.Count > 0)
             {
                 runtimeIdentifiers.AddRange(runtimes.Keys);
-                
+
             }
 
-            if(supports != null && supports.Count > 0)
+            if (supports != null && supports.Count > 0)
             {
                 runtimeSupports.AddRange(supports.Keys);
             }
 
             var union = string.Join(";", runtimeIdentifiers.Union(runtimeSupports));
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                 buildProject.SetProperty("RuntimeIdentifiers", union);
-            });            
+            });
+
         }
 
-        private static void RemoveProjectJsonReference(Microsoft.Build.Evaluation.Project buildProject, string projectJsonFilePath)
+        private static async Task RemoveProjectJsonReferenceAsync(Microsoft.Build.Evaluation.Project buildProject, string projectJsonFilePath)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
-            {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                var projectJsonItem = buildProject.GetItems("None")
+            var projectJsonItem = buildProject.GetItems("None")
                 .Where(t => string.Equals(t.EvaluatedInclude, Path.GetFileName(projectJsonFilePath), StringComparison.OrdinalIgnoreCase))
                 .FirstOrDefault();
 
-                if (projectJsonItem != null)
+            if (projectJsonItem != null)
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate
                 {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                     buildProject.RemoveItem(projectJsonItem);
-                }
-            });
+                });
+
+            }
         }
 
-        private static void CreateBackup(LegacyCSProjPackageReferenceProject project,
+        private static async Task CreateBackupAsync(LegacyCSProjPackageReferenceProject project,
             string projectJsonFilePath)
         {
-            var backupDirectory = Path.Combine(Path.GetDirectoryName(project.MSBuildProjectPath), "Backup");
+            await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate
```

Comment:
This background thread work appears to be all exclusively file I/O. No CPS nor VS interaction. So I don't see any reason for JTF to be used here. How about you remove that, and leave the `await TaskScheduler.Default`?